docs/language.rst

```diff
@@ -125,14 +125,18 @@ Literal values can be assigned to variables
 Primitive types
 ==============================
 
-The basic primitive types are ``string``, ``number``, ``int`` or ``bool``. These basic types also support type casts:
+The basic primitive types are ``string``, ``number``, ``float``, ``int`` or ``bool``. These basic types also support type casts:
+
+.. note::
+    ``number`` will be deprecated in the future
+
 
 .. code-block:: inmanta
 
     assert = true
     assert = int("1") == 1
-    assert = number("1.2") == 1.2
-    assert = number(true) == 1
+    assert = float("1.2") == 1.2
+    assert = float(true) == 1
```

Comment:
```suggestion
    assert = int(true) == 1
```
I would not use `float` for this one:
- it makes little sense, I can't think of any use case for converting a bool to float
- the comparison to `== 1` in a casting section might make the reader believe that you would get exactly that, while you would actually get 1.0.